### ISSUE TITLE
Defines are not loaded when using ES6 modules ("browser" resolution)

### ISSUE THREAD

**Issue by ts-thomas** [google/closure-compiler#3470]
We heavily use closure compiler since years for many projects. A while ago we switched from goog.provide to the new ES6 module system. That is simply the new standard. When I use "browser" resolution to load ES6 modules via the closure compiler the @define flags will not load or will not assign. That makes things a bit complicated in our build stack, because we use several distributed environment files which contains just @define flags. Please improve the support of ES6 modules. It would be really helpful.

Is there already a solution to this issue?

**Comment by EatingW** [comment]
Created internal issue http://b/141321951

**Comment by lauraharker** [comment]
This has been discussed previously at https://github.com/google/closure-compiler/issues/1601.

tl;dr from that issue: if you're using Closure library, `goog.define` works inside a ES module, but the plain `@define` does not :(
